common-validation.lic

```diff
@@ -0,0 +1,39 @@
+# quiet
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_development#common-validation
+=end
+
+class CharacterValidator
+  def initialize(announce, sleep, greet)
+    waitrt?
+    fput('sleep') if sleep
+
+    @lnet = (Script.running + Script.hidden).find { |val| val.name == 'lnet' }
+    @validated_characters = []
+    @greet = greet
+
+    @lnet.unique_buffer.push("chat Lichbot is up and running in room #{Room.current.id}! Whisper me 'help' for more details.") if announce
+  end
+
+  def validate(character)
+    return if valid?(character)
+
+    echo "Attempting to validate: #{character}"
+    @lnet.unique_buffer.push("who #{character}")
+  end
+
+  def confirm(character)
+    return if valid?(character)
+
+    echo "Successfully validated: #{character}"
+    @validated_characters << character
+
+    return unless @greet
+
+    put "whisper #{character} Hi! I'm your friendly neighborhood lichbot. Whisper me 'help' for more details. Don't worry, I've memorized your name so you won't see this message again."
```

Comment:
We might consider making 'lichbot' here configurable so lichbot and bankbot have different strings.

Or maybe (and this could be out of scope of this PR) bankbot can output their balance here, at greeting time.